enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+conflicts when sharing credentials with operands.
+
+Operator authors have a way to notify, guide, and assist OLM Operator admins in providing the required cloud provider
+credentials matched to their permission needs for install and update.
+
+Ideally, a solution here will work in both HyperShift (TAT always) and non-HyperShift(but TAT-enabled) clusters.
+
+### Non-Goals
+
+In-payload/CVO-managed operators are not included in this proposal, they are pre-provisioned for TAT enabled clusters
+(at least for HyperShift)
+
+Bring Your Own Credentials (BYOC) where an operator can manage and distribute credentials to its operands are out of
+scope.
+
+Sharing a set of credentials between all operands to aid multi-tenancy.
+
+## Proposal
+
+**Cloud Credential Operator (CCO) changes**: Adds a token-aware mode, while nominally set in "Manual" mode, CCO will
+look for and process CredentialRequests referenced in Operator CRs when:
+
+```bash
+# infrastructure platform is AWS
+$ oc get infrastructure cluster -o jsonpath --template '{ .status.platform }'
+AWS
+
+# credentialsMode is "Manual"
+$ oc get cloudcredential cluster -o jsonpath --template '{ .spec.credentialsMode }'
+Manual
+
+# serviceAccountIssuer is non empty
+$ oc get authentication cluster -o jsonpath --template='{ .spec.serviceAccountIssuer }'
+abutcher-oidc.s3.us-east-1.amazonaws.com
+```
+
+This will operate in one of two ways depending on which other parts of this EP for other components are adopted: 
+specifically HyperShift/Rosa adding CCO with these token-aware changes or continuing with the current support for the 
+pod identity webhook.
+
+CCO's token-aware mode will work by adding, for example for STS, a "role-arn"-type field on operator added 
+CredentialsRequest objects. This is a new API field on the CredentialsRequest spec. When CCO acquires a 
+CredentialsRequest it will create a Secret (as today mostly) with new functionality: Adding to the Secret a path to the
+projected ServiceAccount token (static for OCP) and a roleARN (from the new field on the CredentialsRequest).
+
+CCO with the changes anticipated here, operating in Manual mode, will continue to ignore CredentialsRequests that do not
+contain the new STS related fields and as these are new fields, existing clusters won't have CredentialsRequests with 
+those fields already.
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+CCO should consider adding the above, three-part test as a utility function exposed in the API.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
```

Comment:
Is this proposing running the CCO in the management side?
Can we specify impact on control plane resource consumption, RBAC, networking access and cloud credentials reqs?

Reply:
CCO will create config secrets in the same namespace where the CredRequest exists, while these are secrets they do not contain sensitive material such as actual credentials, those all go into the bound service account tokens.
CCO will, in the future, potentially assume roles for purposes of validating permissions assigned to a given role match the requirements.
Need to read Infrastructure CR and CloudCrentials CR.

I'm not sure if those require that it runs on management side or not, what do you think?

Reply:
@bentito does the EP actually state where the CCO will run in a hypershift guest cluster topology?

Reply:
No, probably best bet is to ask some from HyperShift team for a suggestion on this. @csrwng can you help?

Reply:
then it sounds like it is not resolved.  but i'll reiterate my expectation that the CCO runs in the control plane NS, not on the customer's worker nodes, and not visible to the customer.

Reply:
I committed changes inline with this threads commentary.